### PR TITLE
Add an example of GCH for molecular materials

### PR DESCRIPTION
**Pull request by ceriottm**
This is a simple but complete example of calculations of the GCH for a dataset of organic crystal polymorphs

### CODE REVIEW

**Comment by PicoCentauri** [comment]
`ase` is not a dependency for the examples. Can you add `ase` to `examples/requirements.txt`?

**Comment by rosecers** [comment]
Can you run the notebook so we can see the outputs?

**Comment by ceriottm** [comment]
Looks like the doc builder does not honor `requirements.txt`

**Comment by PicoCentauri** [comment]
Yes, unfortunately you have to also add this to docs/requirements.txt for now. This, issue will be fixed in #170.

**Comment by ceriottm** [comment]
Also, is everyone happy with having try - except import clauses to show how one could compute features using rascaline?

**Comment by PicoCentauri** [comment]
I would not against using a rascaline in an howto. What about in an extra howto guide? Personally, I don't like putting these `try` `except` blocks in examples. If people want to run the full example they can install these dependencies.

File: examples/GCH-Roy.ipynb [hunk 1]
```diff
@@ -0,0 +1,562 @@
...
+       "ChemiscopeWidget(value='{\"meta\": {\"name\": \" \"}, \"settings\": {\"map\": {\"x\": {\"property\": \"density\"}, \"y\": {\"prop…"
+      ]
+     },
+     "metadata": {},
+     "output_type": "display_data"
+    }
+   ],
+   "source": [
+    "try:\n",
+    "    import chemiscope\n",
```
Comment (PicoCentauri) [inline examples/GCH-Roy.ipynb:245]: I would just let it run without the block. This is the last block of your example and we can add chemiscope as an example dependency.

File: tests/test_datasets.py [hunk 2]
```diff
@@ -105,5 +106,53 @@ def test_datapoint_value(self):
...
+        if self.has_ase is True:
+            self.assertTrue(
+                np.allclose(
+                    self.roy["energies"],
+                    [f.info["energy"] for f in self.roy["structures"]],
+                    rtol=1e-6,
+                )
+            )
+
+
```
Comment (rosecers) [inline tests/test_datasets.py:156]: To make the codecov gods happy, have you tried doing this without explicitly storing `has_ase`?

Comment (ceriottm) [reply tests/test_datasets.py]: voilá. better solution courtesy of GPT4, it simulates the absence of ase.io (and pandas for good measure)

**Comment by rosecers** [comment]
@ceriottm in general this looks great! Do we know if it will render the chemiscopes correctly? The [preview](https://scikit-matter--171.org.readthedocs.build/en/171/read-only-examples/GCH-ROY.html) is showing javascript errors, but it could also be my browser.

**Comment by ceriottm** [comment]
> @ceriottm in general this looks great! Do we know if it will render the chemiscopes correctly? The [preview](https://scikit-matter--171.org.readthedocs.build/en/171/read-only-examples/GCH-ROY.html) is showing javascript errors, but it could also be my browser.

OK, solved I think - hid the cells in nbsphinx, I don't think we can get an embedded chemiscope even though it'd be amazing. In all events, not needed for this PR

**Comment by ceriottm** [comment]
Only problem remaining: I noticed that the torch API documentation gets generated locally, but not on readthedocs. @Luthaf can you help? Then I'd merge so we can move to CUDA and start engaging external devs.

**Comment by ceriottm** [comment]
OK to merge?

File: docs/src/intro.rst [hunk 3]
```diff
@@ -15,7 +15,7 @@ of computational materials science and chemistry.
 
 
 
-- Fingerprint Selection:
+- Features and Samples Selection:
```
Comment (rosecers) [inline docs/src/intro.rst:18]: Was this an intentional change?

File: src/skmatter/sample_selection/_base.py [hunk 4]
```diff
@@ -15,7 +15,7 @@
 
 def _linear_interpolator(points, values):
     """
-    Returns linear interpolater for unstructured D-D data. Tessellate the input point
+    Returns linear interpolator for unstructured D-D data. Tessellate the input point
```
Comment (rosecers) [inline src/skmatter/sample_selection/_base.py:18]: I typically don't like multiple types of changes within one PR, can you roll these into a separate one?

**Comment by rosecers** [comment]
> OK to merge?

Two lingering things then should be good

File: docs/src/intro.rst (hunk 3 above)
Comment (ceriottm) [reply docs/src/intro.rst:18]: Yes - the text and methods below refers to both features and samples, so it seemed more consistent to mention also samples in the title.

File: src/skmatter/sample_selection/_base.py (hunk 4 above)
Comment (ceriottm) [reply src/skmatter/sample_selection/_base.py:18]: I am all for consistency but I'm not going to make a separate pull request to correct a couple of typos.

**Pull request merged**